### PR TITLE
GQL015 Can't infer field name analyzer

### PR DESCRIPTION
**Pull request by gao-artur**
Fixes
 - #3992

### CODE REVIEW

**Comment by gao-artur** [comment]
@Shane32 do you want it in 7.9?

**Comment by Shane32** [comment]
Your call.  It would be useful in 7.9.

**Comment by gao-artur** [comment]
Will do

**Comment by gao-artur** [comment]
Tried to rebase the branch, but something went wrong every time 😞
Will just create a new branch and cherry-pick the code.

**Comment by gao-artur** [comment]
Ok, I think the rebase worked 🎉

File: docs2/site/docs/analyzers/gql015.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# GQL015: Can't infer a Field name from expression
+
+|                        | Value  |
+| ---------------------- | ------ |
+| **Rule ID**            | GQL015 |
+| **Category**           | Usage  |
+| **Default severity**   | Error  |
+| **Enabled by default** | Yes    |
+| **Code fix provided**  | No     |
+| **Introduced in**      | v8.0   |
```
Comment (gao-artur) [inline docs2/site/docs/analyzers/gql015.md:10]:
```suggestion
| **Introduced in**      | v7. 9  |
```

**Review by Shane32 — APPROVED** [review]

**Pull request merged**